modal/image.py

```diff
@@ -638,7 +639,14 @@ def add_local_file(self, local_path: Union[str, Path], remote_path: str, *, copy
         mount = _Mount.from_local_file(local_path, remote_path)
         return self._add_mount_layer_or_copy(mount, copy=copy)
 
-    def add_local_dir(self, local_path: Union[str, Path], remote_path: str, *, copy: bool = False) -> "_Image":
+    def add_local_dir(
+        self,
+        local_path: Union[str, Path],
+        remote_path: str,
+        *,
+        copy: bool = False,
+        ignore: Union[list[str], Callable[[Path], bool]] = [],
```

Comment:
Juxtaposing "copy" and "ignore" like this makes me read `ignore` as wanting a boolean. I guess the type signature is clear though. But we do need to document how the different arguments are interpreted.